### PR TITLE
Add image data structure and matrix multiplication algorithms

### PR DESCRIPTION
**Pull request by IzmaylovI**
1) Added a description of the project in the Readme.md
2) A data structure has been created to store the image in the form (bgr/rgb). Implemented parsing of BMP formats without third-party dependency
3) Matrix multiplication functions have been added using intrinsics NEON and OpenMP

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,21 @@
+CMAKE_MINIMUM_REQUIRED(VERSION 3.25.0)
+
+set(CMAKE_CXX_COMPILER "C:/MinGW/bin/g++.exe")
+set(CMAKE_C_COMPILER "C:/MinGW/bin/gcc.exe")
```
Comment (aobolensk) [inline CMakeLists.txt]: This solution is not universal, at least do not override C/CXX compiler if it is already found

File: headers/Color.hpp [hunk 2]
```diff
@@ -0,0 +1,163 @@
+#ifndef _COLOR_
+#define _COLOR_
+#include <iostream>
+enum mode { rgb, bgr };
```
Comment (aobolensk) [inline headers/Color.hpp]: It is not the mode in fact, it could be called "layout"

File: headers/Function.hpp [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+    omp_set_num_threads(thr);
+#pragma omp parallel for collapse(cllps)
+    for (int i = 0; i < row1; ++i)
+        for (int j = 0; j < col2; ++j)
+            for (int k = 0; k < col1; ++k)
+                res[i * col2 + j] += data1[i * col1 + k] * data2[k * col2 + j];
+    return res;
+}
+
+/*
```
Comment (aobolensk) [inline headers/Function.hpp]: Do we need this comment?

File: headers/Structer.hpp [hunk 4]
```diff
@@ -0,0 +1,239 @@
...
+
+template <typename ValType>
+class Mtrx {
+protected:
+    ValType* data = nullptr;
+public:
+    int height;
+    int width;
+
+    Mtrx(int height_m = 0, int width_m = 0, ValType element = 0);
```
Comment (aobolensk) [inline headers/Structer.hpp]: Why do we have default width/height set to 0?

**Comment by IzmaylovI** [comment]
Fixed the code:
1) removed the override C/CXX compiler.
2) renamed mode to layout
3) removed the commentaries in the file Function.hpp. For matrices of elements of the base type, everything will work. For pixel matrices, one more function needs to be implemented
4) removed the default value of 0 in width/height because there is no need for this

File: CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+include_directories(./headers)
+
+find_package(OpenMP)
+if(OpenMP_FOUND)
+    message(STATUS "Link OpenMP")
+else()
+    message(STATUS "Not link OpenMP")
+endif()
+
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${OpenMP_C_FLAGS}")
```
Comment (allnes) [inline CMakeLists.txt]: please set OpenMP's flags when `OpenMP_FOUND` is true

File: README.md [hunk 6]
```diff
@@ -1 +1,19 @@
...
+# itlab_2022
+
+# low-level optimization of computer vision algorithms on ARM architectures     <img src="https://user-images.githubusercontent.com/75639514/229908421-84a5837b-4220-423a-8cc3-ab97e548a077.png" width="35" height="35">
+## Objectives of this project:
+- Study of classical computer vision algorithms, as well as work with artificial neural networks;
+- Optimization of these algorithms on ARM architectures. The study of parallel programming. Using Vector instructions;
+- Working with Raspberry PI devices in practice.
+
+## Software modules:
+### Module 1.
```
Comment (allnes) [inline README.md]: Please name the modules according to their purpose

File: headers/Structer.hpp [hunk 7]
```diff
@@ -0,0 +1,239 @@
+#ifndef _MTRX_
+#define _MTRX_
+
+#include "Color.hpp"
+
+template <typename ValType>
+class Mtrx {
```
Comment (allnes) [inline headers/Structer.hpp:8]: I think this class should be made purely abstract. 
@aobolensk how do you think?

**Review by allnes — CHANGES_REQUESTED** [review]

**Comment by allnes** [comment]
@IzmaylovI please correct commit changes, old changes is lost

**Comment by IzmaylovI** [comment]
Corrected the commit. Is there anything else that needs to be corrected?

**Review by allnes — APPROVED** [review]

**Comment by aobolensk** [comment]
Please, update the title and provide description. I think it is better to squash this pull request and we need to preserve quality information in main branch commit. Other things LGTM

**Comment by aobolensk** [comment]
@allnes please, use squash strategy when we will be merging this PR

**Comment by IzmaylovI** [comment]
> Please, update the title and provide description. I think it is better to squash this pull request and we need to preserve quality information in main branch commit. Other things LGTM


where do we need to change the description?

**Comment by aobolensk** [comment]
> > Please, update the title and provide description. I think it is better to squash this pull request and we need to preserve quality information in main branch commit. Other things LGTM
> 
> 
> where do we need to change the description?

Description of this pull request

**Comment by IzmaylovI** [comment]
1) Removed extra lines in CMakeLists.txt
2) Wrote non-template matrix multiplication functions using intrinsics
3) Тame the modules according to their purpose in Readme.md
4) Set OpenMP's flags when "OpenMP_FOUND" is true  in CMakeLists.txt
5) Deleted unnecessary file - BMP.cpp 
6) Removed unnecessary comments in files

**Review by aobolensk — APPROVED** [review]

**Pull request merged**